mathesar/api/serializers/schemas.py

```diff
@@ -30,7 +33,30 @@ class Meta:
         ]
 
     def get_num_tables(self, obj):
-        return obj.tables.count()
+        confirmed_table_roles = (Role.EDITOR.value, Role.VIEWER.value)
+        user = self.context['request'].user
+
+        confirmed_table = "CONFIRMED"
+        unconfirmed_table = "UNCONFIRMED"
+        not_allowed = "NOT_ALLOWED"
+
+        if not user.is_superuser:
+            permissible_schema_role_filter = Q(schema__schema_role__role__in=confirmed_table_roles)
+            permissible_table_view_filter = Q(import_verified=True) | Q(import_verified__isnull=True)
+
+            qs = obj.tables.filter(schema__schema_role__user=user).annotate(
+                to_count=Case(
+                    When(permissible_schema_role_filter, then=Case(
+                        When(permissible_table_view_filter, then=Value(confirmed_table)),
+                        default=Value(unconfirmed_table)
+                    )),
+                    default=Value(not_allowed), output_field=django.db.models.CharField()
+                )
+            )
+            count = qs.filter(Q(to_count=not_allowed) | Q(to_count=confirmed_table)).count()
+        else:
+            count = obj.tables.count()
+        return count
```

Comment:
From what I could tell, the `get_table_list()` function in that gets called to fetch the table list returns an empty list in case the schema is not supplied to the function. The view associated with `/<db_name>/` supplied no schema to the `get_common_data()` function and consequently `get_table_list()` gets no schema either, this results in an empty array. The `/<db_name>/<id>` endpoint on the other hand does supply a schema which returns unconfirmed and confirmed or only confirmed tables of that schema based on the user schema role. So this view does use the updated `scope_queryset()` logic to return the required tables.

So do I need to modify the `/<db_name>/` endpoint to return the required tables across all schemas?

Reply:
Seems like the common_data dict is already using the `scope_queryset()`. No need to change anything there.

Reply:
@DankMemes4President Please reuse the `scope_queryset()` of the `TableAccessPolicy` here. The last comment was meant for `common_data` dict.

Reply:
Yep, this is done as well.